### PR TITLE
PLF-7270 : [Enable/Disable user feature]Filter doesn't work

### PR DESCRIPTION
**Pull request by aboughzela**

### CODE REVIEW

File: component/organization/pom.xml [hunk 1]
```diff
@@ -124,6 +124,14 @@
       <artifactId>mockito-all</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.exoplatform.commons</groupId>
+      <artifactId>commons-component-upgrade</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.exoplatform.commons</groupId>
+      <artifactId>commons-component-upgrade</artifactId>
```
Comment (boubaker) [inline component/organization/pom.xml]: Duplicated declaration

File: component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+                StringBuilder temp= new StringBuilder();
+                int i = 0;
+
+                while (rs.next()) {
+                    i++;
+                    temp.append(rs.getString(COLUMN_ID));
+                    if(i% 1000 == 0){
+                        removeBatch(connection, temp.toString());
+                        connection.commit();
+                        LOG.info("EnableUserUpgradePlugin : clean in progress , {} done", i);
```
Comment (boubaker) [inline component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java]: Could you please display the total count too (use a dedicated query to count the items)

File: component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Upgrade class used to remove IDM attribute enabled (From table jbid_io_attr ) when value equal to TRUE
+ * (From table jbid_io_attr_text_values)
+ */
+public class EnableUserUpgradePlugin extends UpgradeProductPlugin {
```
Comment (boubaker) [inline component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java]: Please add a unit test for this class

File: component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+            }
+        } catch (SQLException e) {
+            LOG.error("Error while Clean items", e);
+        } finally {
+            picketLinkIDMCacheService.invalidateAll();
+            if(organizationService!= null && organizationService instanceof PicketLinkIDMOrganizationServiceImpl){
+                ((PicketLinkIDMOrganizationServiceImpl)organizationService).clearCaches();
+            }
+            if (success){
+                LOG.info("EnableUserUpgradePlugin : finished successfully, Clean items {}",results.size());
```
Comment (boubaker) [inline component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java]: Could you add a log when the processing is finished, with the elapsed time

File: extension/config/src/main/resources/conf/platform/upgrade-sample.properties [hunk 5]
```diff
@@ -54,3 +54,8 @@ commons.upgrade.SpaceMemberAnyMembershipUpgradePlugin.enable=true
 #PLF-7169 : Upgrade from ISPN 5 to ISPN 8
 org.exoplatform.jcr.locks.force.remove=true
 #To upgrade to ISPN 8 , you need to clean old lock entries from cache store.
+
+# PLF-7270 [Enable/Disable user feature]Filter doesn't work after migration.
+# Remove IDM attribute enabled (From table jbid_io_attr ) when value equal to TRUE
+# (From table jbid_io_attr_text_values)\
+org.exoplatform.platform.migration.EnableUserUpgradePlugin=true
```
Comment (boubaker) [inline extension/config/src/main/resources/conf/platform/upgrade-sample.properties]: You should redefine the method isEnabled in EnableUserUpgradePlugin to force it to be enabled on startup. So you an delete this upgrade plugin declaration here.

File: component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java (hunk 2 above)
Comment (aboughzela) [reply component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java]: Done

File: component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java (hunk 4 above)
Comment (aboughzela) [reply component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java]: Done

File: extension/config/src/main/resources/conf/platform/upgrade-sample.properties (hunk 5 above)
Comment (aboughzela) [reply extension/config/src/main/resources/conf/platform/upgrade-sample.properties]: Done

File: component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java (hunk 3 above)
Comment (aboughzela) [reply component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java]: IMO this upgrade should validated on TQA and Tribe
What you think ?

Comment (boubaker) [reply component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java]: Of course, but the Unit test remain necessary to be sure for future changes that this remain functional.

Comment (aboughzela) [reply component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java]: Unit test Added

File: component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+                connection.setAutoCommit(false);
+                countItemsStatement = connection.prepareStatement(SELECT_COUNT_ENABLE_ATTRIBUTE);
+                count = countItemsStatement.executeQuery();
+                if (count.next() && count.getInt(1) > 0){
+                    nb = count.getInt(1);
+
+                    findItemsStatement = connection.prepareStatement(SELECT_ENABLE_ATTRIBUTE);
+                    findItemsStatement.setFetchSize(1000);
+
+                    LOG.info("EnableUserUpgradePlugin : Start Select items NAME=enabled and ATTR_VALUE=true");
```
Comment (vsellier) [inline component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java]: No needs to add ''EnableUserUpgradePlugin : '' on each log as it's present on the logs via the logger name

File: component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+                    StringBuilder temp= new StringBuilder();
+                    int i = 0;
+
+                    while (rs.next()) {
+                        i++;
+                        temp.append(rs.getString(COLUMN_ID));
+                        if(i% 1000 == 0){
+                            removeBatch(connection, temp.toString());
+                            connection.commit();
+                            LOG.info("EnableUserUpgradePlugin : clean in progress , {}/{} done", i, nb);
```
Comment (vsellier) [inline component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java]: WDYT of "clean in progress : {}/{}" ?

File: component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+                public Connection execute(Connection conn) throws SQLException {
+                    return conn;
+                }
+            });
+            if (connection != null) {
+                auto = connection.getAutoCommit();
+                connection.setAutoCommit(false);
+                countItemsStatement = connection.prepareStatement(SELECT_COUNT_ENABLE_ATTRIBUTE);
+                count = countItemsStatement.executeQuery();
+                if (count.next() && count.getInt(1) > 0){
```
Comment (vsellier) [inline component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java]: Bracket indentation

File: component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+                    rs = findItemsStatement.executeQuery();
+
+
+                    StringBuilder temp= new StringBuilder();
+                    int i = 0;
+
+                    while (rs.next()) {
+                        i++;
+                        temp.append(rs.getString(COLUMN_ID));
+                        if(i% 1000 == 0){
```
Comment (vsellier) [inline component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java]: check indentation

File: component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
+        try {
+            connection = session.doReturningWork(new ReturningWork<Connection>() {
+                @Override
+                public Connection execute(Connection conn) throws SQLException {
+                    return conn;
+                }
+            });
+            if (connection != null) {
+                auto = connection.getAutoCommit();
+                connection.setAutoCommit(false);
```
Comment (vsellier) [inline component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java]: is it good to manipulate the autocommit status of the connection here ?

File: component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java [hunk 11]
```diff
@@ -0,0 +1,204 @@
...
+                            LOG.info("EnableUserUpgradePlugin : clean in progress , {}/{} done", i, nb);
+                            temp = new StringBuilder();
+                        }else{
+                            temp.append(",");
+                        }
+                    }
+                    if(i% 1000 != 0){
+                        removeBatch(connection, temp.substring(0, temp.lastIndexOf(",")));
+                        connection.commit();
+                        LOG.info("EnableUserUpgradePlugin : clean in progress , {}/{} done", i, nb);
```
Comment (vsellier) [inline component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java]: same remark as the previous log

File: component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java [hunk 12]
```diff
@@ -0,0 +1,204 @@
...
+                picketLinkIDMCacheService.invalidateAll();
+            }
+            if(organizationService!= null && organizationService instanceof PicketLinkIDMOrganizationServiceImpl){
+                ((PicketLinkIDMOrganizationServiceImpl)organizationService).clearCaches();
+            }
+            if (success){
+                LOG.info("EnableUserUpgradePlugin : finished successfully  on {} milliseconds, Clean {} items",(System.currentTimeMillis()-started), nb);
+            } else {
+                LOG.warn("EnableUserUpgradePlugin : An unexpected error occurs when migrating");
+                throw new RuntimeException("Error during remove enable attribute");
```
Comment (vsellier) [inline component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java]: If an exception is thrown here, the remaining actions of the finally will not be executed

File: component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java (hunk 10 above)
Comment (aboughzela) [reply component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java]: I need to disable auto commit to  execute atomic operation (removeBatch)

File: component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java (hunk 8 above)
Comment (aboughzela) [reply component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java]: done

File: component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java (hunk 6 above)
Comment (aboughzela) [reply component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java]: done

File: component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java (hunk 9 above)
Comment (aboughzela) [reply component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java]: done

File: component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java (hunk 7 above)
Comment (aboughzela) [reply component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java]: yes it's more clean

File: component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java (hunk 11 above)
Comment (aboughzela) [reply component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java]: done

File: component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java (hunk 12 above)
Comment (aboughzela) [reply component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java]: good catch , I moved it

File: component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java (hunk 10 above)
Comment (vsellier) [reply component/organization/src/main/java/org/exoplatform/platform/migration/EnableUserUpgradePlugin.java]: 👍, I was going to ask why not to using the batch api of jdbc, but the behavior around autocommit seems to be dependent of the jdbc implementation, so It looks better to secure it.

**Pull request merged**